### ISSUE TITLE
DL3016 "Pin versions in npm" misfires on flag args

### ISSUE THREAD

**Issue by DanBurton** [hadolint/hadolint#515]
- [x] This is a bug report
- [x] I searched existing issues before opening this one

### Expected behavior

When I use `npm install --loglevel verbose` in a Dockerfile, hadolint should not complain about DL3016.

### Actual behavior

It does.

### Steps to reproduce the behavior

```bash
cat <<eof > Dockerfile
FROM ubuntu@20.04
RUN npm install --loglevel verbose
eof

hadolint Dockerfile
```

Has output:

```bash
Dockerfile:2 DL3016 Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`
```

**Output of `hadolint --version`:**

```bash
$ hadolint --version
Haskell Dockerfile Linter 1.18.0-no-git
```

**Dockerfile**

```Dockerfile
FROM ubuntu@20.04
RUN npm install --loglevel verbose
```

**Additional environment details (OS, stack version, etc.)**
macos (probably irrelevant)

**Comment by WillBartee** [comment]
This issue still occurs for other allowed flags for `npm install`.

### Steps to reproduced the behavior
```shell
cat <<eof > Dockerfile
FROM node:alpine
RUN npm install --global nodemon@"2.0.16" --registry https://registry.npmjs.org/
eof

hadolint Dockerfile
```

**Has output**
```shell
Dockerfile:2 DL3016 warning: Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`
```

**Output of hadolint --version:**

```shell
$ hadolint --version
Haskell Dockerfile Linter 2.10.0
```

**Dockerfile**
```Dockerfile
FROM node:alpine
RUN npm install --global nodemon@"2.0.16" --registry https://registry.npmjs.org/
```
